sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/expressions.scala

```diff
@@ -410,37 +410,41 @@ object SimplifyBinaryComparison
   extends Rule[LogicalPlan] with PredicateHelper with ConstraintHelper {
 
   private def canSimplifyComparison(
-      plan: LogicalPlan, left: Expression, right: Expression): Boolean = {
+      left: Expression,
+      right: Expression,
+      getNotNullExpressions: () => ExpressionSet): Boolean = {
```

Comment:
can we use byname parameter? `notNullExpressions: => ExpressionSet`

Reply:
Got it, updated.